### ISSUE TITLE
Is it possible? Detection fails depending on PS version in system context?

### ISSUE THREAD

**Issue by KnifMelti** [Romanitho/Winget-Install#22]
As title says (when running in an admin user context there's no problem) ...

...with https://www.nirsoft.net/utils/advanced_run.html
Run (with installed TimKosse.FileZilla.Client):
-AppIDs TimKosse.FileZilla.Client -Uninstall -WAUWhiteList
(TimKosse.FileZilla.Client-uninstall.ps1 takes care of the whait for the graphical uninstall UI)

Results:
Win 11 PS: 5.1.22000.1 (HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine)
```
20:27:21 - Using following Winget Cmd: C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10271.0_x64__8wekyb3d8bbwe\winget.exe (v1.2.10271)
/../
20:27:45 - TimKosse.FileZilla.Client successfully uninstalled.
```

Win 10 PS: 5.1.19041.1 (HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\PowerShell\3\PowerShellEngine)
```
20:40:26 - Using following Winget Cmd: C:\Program Files\WindowsApps\Microsoft.DesktopAppInstaller_1.17.10271.0_x64__8wekyb3d8bbwe\winget.exe (v1.2.10271)
/../
20:40:39 - TimKosse.FileZilla.Client is not installed.
```

**And, it's not only TimKosse.FileZilla.Client.. ..it's the detection itself and goes for every application detection on my Win 10 with Powershell 5.1.19041.1...**

**Comment by KnifMelti** [comment]
...haven't tried yet with this PS version in real (under SCCM).

**Comment by KnifMelti** [comment]
It's not a scripting issue.

**Comment by KnifMelti** [comment]
https://github.com/microsoft/winget-cli/discussions/2096

**Comment by Romanitho** [comment]
Same here on some computers. Sometimes if I uninstall, reinstall Visual C++ and restart it works. But not sure. Maybe it's due to something else.

**Comment by KnifMelti** [comment]
Which one (of the 24 installed) Microsoft Visual C++ Redistributable (developing machine) do you propose I should uninstall/install then ;)
Which one do you choose when you have the issue?

**Comment by Romanitho** [comment]
My recent tests was on a fresh install.
When I install 2022 as system and reinstall it as admin user. But I think its not really related to this. Not sure

**Comment by KnifMelti** [comment]
I uninstalled all 2013, didn't reboot and suddenly it did work one time!
Tested one more time and it failed...
Rebooted and still failing...
...so, related/unrelated - it's something...

**Comment by Romanitho** [comment]
One interesting thing : If I update winget with their latest preview release, it works again ^^
https://github.com/microsoft/winget-cli/releases/tag/v1.3.431

**Comment by KnifMelti** [comment]
Then it's the classical Microsoft **_wait..._** ;)

**Comment by Romanitho** [comment]
I think i'll use the preview one in production as it looks more stable than the current one ^^

**Comment by KnifMelti** [comment]
Then you've have to install it on every client first...

**Comment by Romanitho** [comment]
yes. I was thinking about detecting Winget version on WAU first install. Then, put WAU as prerequisites when I install apps with Intune or SCCM for example.

**Comment by KnifMelti** [comment]
Yes, a prerequisites dep. to WAU is on my mind also!

**Comment by Romanitho** [comment]
I have a PR waiting for publishing in WAU repo :)
And for dependencies, normally everything is (or will be) in the WAU install script.
In that case, I only put WAU as a prerequisite of my apps installed with Winget-Install script

**Comment by KnifMelti** [comment]
Could you also incorporate the Mods solution for WAU it would be nice:
https://github.com/Romanitho/Winget-AutoUpdate/issues/42
With **$ModsInstal**l both for Upgrade/Install

**Comment by Romanitho** [comment]
we can create a separate PR for that.
by the way, how do you handle your "AppID_upgrade.ps1" mod files ?

**Comment by KnifMelti** [comment]
The same way as in winget-install.ps1:
```
        #Check if mods exist
        $ModsInstall = Test-ModsInstall $AppID
        if ($ModsInstall){
            Write-Log "Modifications for $AppID during install are being applied..." "Yellow"
            & "$ModsInstall"
        }
```
I'll create an PR out of Issue #42 (not now, I'm too busy) :)

**Comment by KnifMelti** [comment]
Ah, https://github.com/Romanitho/Winget-AutoUpdate/discussions/57...

**Comment by KnifMelti** [comment]
WAU Mods done.
Winget latest preview release solution to the crash in winget list/export

**Comment by EricPez** [comment]
hello I am having exactly the same issue. Do someone has already found a way to fix that ? or shall we have to wait on Microsoft to release a fix with an upcoming Windows Update ?

**Comment by KnifMelti** [comment]
> hello I am having exactly the same issue. Do someone has already found a way to fix that ? or shall we have to wait on Microsoft to release a fix with an upcoming Windows Update ?

Winget latest preview release solution to the crash in winget list/export
